jsbits/diff.js

```diff
@@ -198,168 +205,152 @@ function syncChildren (os, ns, parent) {
-		    if (last > newNodeIndex) moved = true;
-		    /* Update last seen */
-		    last = newNodeIndex;
-		}
-	    }
-	    /* If `moved` flag is on, or if the length of the old children list minus the number of
-	       removed nodes isn't equal to the length of the new children list. Then go to the next step */
-	    if (moved /* DMJ: not sure we need this predicate ? ===> */ || (oLen - removedNodes !== nLen)) {
-		/* Find minimum number of moves if `moved` flag is on, or insert new nodes if the length is changed. */
-		LIS = lis(P);
-		lisIndex = LIS.length - 1;
-		while (nLen > -1) {
-		    if (P[lisIndex] === nLen) {
-			listIndex--;
-			continue;
-		    }
-		    else if (LIS[lisIndex] !== nLen) {
-			/* node has moved */
-			ns[nLen].domRef = os[P[nLen]].domRef;
-			parent.insertBefore(ns[nLen].domRef, os[nLen].domRef);
-		    }
-		    nLen--;
-		}
-	    } else if (!moved) {
-		/* When moved flag is off, we don't need to find LIS, and we just
-		   iterate over the new children list and check its
-		   current position in the positions array, if it is `-1`,
-		   then we insert new node. */
-		for (i = newFirstIndex; i <= newLastIndex; i++) {
-		    if (P[i] === -1) {
-			createNodeDontAppend(ns[i]);
-			/* Replace whatever is at current */
-			/* parent.replaceChild(ns[i].domRef, parent.children[i]); ... this doesn't seem right to me... */
-			parent.insertBefore(ns[i].domRef, parent.children[i+1]);
-		    }
-		}
+	    /* final case, perform linear search to check if new key exists in old map, decide what to do from there */
+	    found = false;
+	    tmp = oldFirstIndex;
+	    while (tmp <= oldLastIndex) {
+       	      if (os[tmp].key === nFirst.key) {
+		  found = true;
+		  node = os[tmp];
+		  break;
+	      }
+              tmp++;
 	    }
-	}
-    }
-}
-
-
-/* Thanks Boris :) */
-function lis(a) {
-  var p = a.slice(0);
-  var result = [], u, v, il, c, j;
-  result.push(0);
-
-  for (var i = 0, il = a.length; i < il; i++) {
-    if (a[i] === -1) {
-      continue;
-    }
-
-    j = result[result.length - 1];
-    if (a[j] < a[i]) {
-      p[i] = j;
-      result.push(i);
-      continue;
-    }
+	    /* If new key was found in old map this means it was moved, hypothetically as below
+	       -> [ a e b c ] <- old children
+	       -> [ b e a j ] <- new children
+                    ^
+	       In the above case 'b' has been moved, so we need to insert 'b' before 'a' in both vDOM and DOM
+	       We also increase oldFirstIndex and newFirstIndex.
 
-    u = 0;
-    v = result.length - 1;
+	       This results in new list below w/ updated index position
+	       -> [ b a e c ] <- old children
+	       -> [ b e a j ] <- new children
+                      ^
+   	    */
+	    if (found) {
+		/* Swap DOM references */
+		parent.insertBefore(node.domRef, oFirst.domRef);
+                /* Move item to correct position */
+        	os.splice(oldFirstIndex, 0, os.splice(tmp, 1)[0]);
```

Comment:
And here.

Reply:
```javascript
Array.prototype.move = function(from, to) {
    this.splice(to, 0, this.splice(from, 1)[0]);
};
```

Reply:
I still suspect that this is potentially slower than
```js
Array.prototype.move = function(from, to) {
  var tmp = this[to];
  this[to] = this[from];
  this[from] = tmp;
}
```

Reply:
https://jsperf.com/ it :)

Reply:
It’s not even close, using a temporary variable is clearly faster (the difference is similar in Chrome)
![screenshot-2017-9-25 swap two elements in an array jsperf](https://user-images.githubusercontent.com/1313584/30793418-20cc0638-a1c2-11e7-9140-beaf443b0f7a.png)

Reply:
Very nice. I'm convinced. Can you make a PR?